File: 2550. Count Collisions of Monkeys on a Polygon/Solution.py
class Solution:
    def monkeyMove(self, n: int) -> int:
        high, MOD, res = self.MSB(n) >> 1, 1000000007, 2
        while high > 0:
            res = (res * res) % MOD
            if (high & n) != 0: res *= 2
            high >>= 1
        return ((res + MOD) - 2) % MOD

    def MSB(self, n):
        l = -1
        while n > 0:
            n >>= 1
            l += 1
        return 1 << l

File: 2550. Count Collisions of Monkeys on a Polygon/Solution.java
class Solution {
    public int monkeyMove(int n) {
        int high = Integer.highestOneBit(n);
        long MOD = 1000000007, e = 2;
        for (high >>= 1; high > 0; high >>= 1) {
            e = (e * e) % MOD;
            if ((high & n) != 0) e = (e * 2);
        }
        return (int) (((e + MOD) - 2) % MOD);
    }
}